public/app/features/plugins/admin/pages/Browse.tsx

```diff
@@ -29,8 +31,9 @@ export default function Browse({ route }: GrafanaRouteComponentProps): ReactElem
   const filterByType = (query.filterByType as string) ?? 'all';
   const sortBy = (query.sortBy as string) ?? 'nameAsc';
 
-  const { plugins, isLoading, error } = usePluginsByFilter({ searchBy: q, filterBy, filterByType });
-  const sortedPlugins = plugins.sort(sorters[sortBy]);
+  const filteredPlugins = useSelector(find(q, filterBy, filterByType));
+  const { isLoading, error } = useFetchStatus();
+  const sortedPlugins = filteredPlugins.sort(sorters[sortBy]);
```

Comment:
Would it make sense to introduce a `useGetAllAndFilter` hook to make it a little bit easier to follow flow of the information?

I get that the `useGetAll()` will trigger a async call to fetch the plugins. Then we use the `useFetchStatus()` to get the status of the current ongoing request. We use the `useSelector(find(...))` to get filtered plugins that we later sort.

So my question is: would it be easier to follow if we did something like the following?

```ts
const filter = find(q, filterBy, filterByType);
const sorter = sorters[sortBy];
const { isLoading, error, plugins } = useFetchPluginsAsync(filter, sorter);
```

Reply:
My thinking is that the `useGetAll` only seems to be used in this place so it can then be replaced by a more specific hook that reads a little bit easier.

Reply:
I like those ideas, it could make it easier to read and reason about as well.

Although I am not against the new `useGetAllAndFilter()` hook, just bringing up another idea, what do you think about something like the following? Or do you find the separate new hook more straightforward?
```javascript
const { isLoading, error, plugins }  = useGetAll({ filters });
```

Reply:
I think that looks great! I just hoped that it would make the code easier to read and reason about. So I am super happy with your suggestion!